Source/PluginProcessor.cpp

```diff
@@ -304,14 +312,14 @@ SynthesiserSound::Ptr SoomplerAudioProcessor::getSampleData(std::optional<File>
         return nullptr;
     }
 
-    auto formatReader = formatManager.createReaderFor(*sampleFile);
+    auto formatReader = formatManager.createReaderFor(sampleFile.value());
 
     setTransportSource(formatReader);
-    thumbnail.setSource(new FileInputSource(*sampleFile));
+    thumbnail.setSource(new FileInputSource(sampleFile.value()));
 
     BigInteger midiNotes;
     midiNotes.setRange(0, 126, true);
-    return new soompler::ExtendedSound(sampleFile->getFileName(), *formatReader, midiNotes, 0x40, 0.0, 0.0, 400.0);
+    return new soompler::ExtendedSound(sampleFile->getFileName(), *formatReader, midiNotes, 0x48, 0.0, 0.0, Settings::MAX_SAMPLE_LENGTH);

```

Comment:
Maybe we need factory method or something. Or pack arguments to a structure. 3-4 arguments is maximum. And if you do some factory then it will be clear what arguments (by function names) you provide to constructor. Now you can't say on invocation side what you pass to a constructor.

Reply:
I extracted arguments to constants, so every argument has name. Maybe in future if there will be need for any extra argument I consider refactor this piece of code.